pkg/deploy/strategy/support/lifecycle.go

```diff
@@ -472,7 +472,7 @@ func NewPodWatch(client kclient.PodsNamespacer, namespace, name, resourceVersion
 		},
 	}
 
-	queue := cache.NewFIFO(cache.MetaNamespaceKeyFunc)
+	queue := cache.NewResyncableFIFO(cache.MetaNamespaceKeyFunc)
```

Comment:
> I don't think we need relist to work for hook pods and since you are breaking an extended test (not sure why) I would say it's fine to move this back to NewFIFO.

Messed up the "addIfNotPresent" since the detection algorithm changed.